debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitmqStreamChangeConsumer.java

```diff
@@ -0,0 +1,161 @@
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.engine.DebeziumEngine.RecordCommitter;
+import io.debezium.engine.Header;
+import io.debezium.server.BaseChangeConsumer;
+
+/**
+ * Implementation of the consumer that delivers the messages into RabbitMQ Stream destination.
+ *
+ * @author Olivier Boudet
+ *
+ */
+@Named("rabbitmq")
+@Dependent
+public class RabbitmqStreamChangeConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitmqStreamChangeConsumer.class);
+
+    private static final String PROP_PREFIX = "debezium.sink.rabbitmq.";
+    @ConfigProperty(name = PROP_PREFIX + "host")
+    String host;
+
+    @ConfigProperty(name = PROP_PREFIX + "port", defaultValue = "5672")
+    int port;
+
+    @ConfigProperty(name = PROP_PREFIX + "auth.enabled", defaultValue = "true")
+    boolean authEnabled;
+
+    @ConfigProperty(name = PROP_PREFIX + "auth.username", defaultValue = "guest")
+    String username;
+
+    @ConfigProperty(name = PROP_PREFIX + "auth.password", defaultValue = "guest")
+    String password;
+
+    @ConfigProperty(name = PROP_PREFIX + "virtualHost", defaultValue = "/")
+    String virtualHost;
+
+    @ConfigProperty(name = PROP_PREFIX + "routingKey", defaultValue = "")
+    Optional<String> routingKey;
+
+    Connection connection;
+
+    Channel channel;
+
+    @PostConstruct
+    void connect() {
+        ConnectionFactory factory = new ConnectionFactory();
+        factory.setHost(host);
+        factory.setPort(port);
+        if (authEnabled) {
+            factory.setUsername(username);
+            factory.setPassword(password);
+        }
+        factory.setVirtualHost(virtualHost);
+
+        LOGGER.info("Using connection to {}:{}", host, port);
+
+        try {
+            connection = factory.newConnection();
+            channel = connection.createChannel();
+            channel.confirmSelect();
+        }
+        catch (IOException | TimeoutException e) {
+            throw new DebeziumException(e);
+        }
+    }
+
+    @PreDestroy
+    void close() {
+
+        try {
+            if (channel != null) {
+                channel.close();
+            }
+            if (connection != null) {
+                connection.close();
+            }
+        }
+        catch (IOException | TimeoutException e) {
+            throw new DebeziumException(e);
+        }
+
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+            final String exchange = record.destination();
+
+            try {
+                channel.basicPublish(exchange, routingKey.orElse(""),
```

Comment:
I think if the routing key is not set, then the topic name should be used, WDYT?

Reply:
ok, I changed to use the topic name. But I need to look at stream-mapper-name.